src/WebJobs.Script/Utility.cs

```diff
@@ -996,6 +997,58 @@ public static FunctionAppContentEditingState GetFunctionAppContentEditingState(I
             }
         }
 
+        /// <summary>
+        /// Returns scale monitors and target scalers we want to use based on the configuration.
+        /// Scaler monitor will be ignored if a target scaler is defined in the same extensions assembly and TBS is enabled.
+        /// </summary>
+        /// <param name="environment">Environment variables.</param>
+        /// <param name="hostingConfiguration">Hosting configuration.This is used to enable TDS on stamp level for specific triggers.</param>
+        /// <param name="scaleMonitors">Registered scale monitors.</param>
+        /// <param name="targetScalers">Registered target scalers.</param>
+        /// <param name="scaleMonitorsToSample">Scale monitor to process.</param>
+        /// <param name="targetScalersToSample">Target scaler to process.</param>
+        public static void GetScalersToSample(
+            IEnvironment environment,
+            IFunctionsHostingConfiguration hostingConfiguration,
+            IEnumerable<IScaleMonitor> scaleMonitors,
+            IEnumerable<ITargetScaler> targetScalers,
+            out List<IScaleMonitor> scaleMonitorsToSample,
+            out List<ITargetScaler> targetScalersToSample)
+        {
+            scaleMonitorsToSample = new List<IScaleMonitor>();
+            targetScalersToSample = new List<ITargetScaler>();
+
+            // Check if TBS enabled on app level
+            if (environment.IsTargetBasedScalingEnabled())
+            {
+                HashSet<string> targetScalerAssemblies = new HashSet<string>();
+                foreach (var scaler in targetScalers.Where(x => x != null))
+                {
+                    string assemblyName = scaler.GetType().Assembly.GetName().Name;
+                    string flag = hostingConfiguration.GetValue(assemblyName, null);
```

Comment:
I recommend moving this check into the Environment extensions - so `IsTargetBasedScalingEnabled` will have logic to check both app setting and hosting configuration

Reply:
We decided to implement the checks (app setting and hosting config) is in the function host to avoid duplication of the code in all the extensions - extensions just need to implement the required interface to support TBS.

Also SC will have his own hosting configuration check implementation as it reads hosting config directly from SQL.